src/sdk/pynni/nni/compression/torch/builtin_pruners.py

```diff
@@ -471,3 +490,239 @@ def _get_distance_sum(self, weight, in_idx, out_idx):
 
     def update_epoch(self, epoch):
         self.mask_calculated_ops = set()
+
+
+class ActivationRankFilterPruner(Pruner):
+    """
+    A structured pruning base class that prunes the filters with the smallest
+    importance criterion in convolution layers to achieve a preset level of network sparsity.
+    Hengyuan Hu, Rui Peng, Yu-Wing Tai and Chi-Keung Tang,
+    "Network Trimming: A Data-Driven Neuron Pruning Approach towards Efficient Deep Architectures", ICLR 2016.
+    https://arxiv.org/abs/1607.03250
```

Comment:
It seems this docstring should be moved into subclasses.

Reply:
subclasses also have this docstring